### DISCUSSION TITLE
option to update repos sequentially instead of parallelly

### DISCUSSION THREAD

**Discussion by CaiMiao** [wstyres/Zebra#1488] [Ideas]
**Is your feature request related to a problem? Please describe.**
Yes. I live in China and use own overseas server for Proxy services in dedicated protocols.
And for some unknown reason, the VPN apps (specifically for the Vmess protocol and maybe Trojan-GFW protocol) on App Store consumes much more RAMs in jailbroken devices, leads to a `jetsam` kill when little excessive network load happens.
Even I have manually doubled the default value in my jetsam config, the parallel repo update still triggers jetsam kill easily, making my repo update all failed.

**Describe the solution you'd like**
Provide an option to let repo updates go sequentially,

**Describe alternatives you've considered**
or parallelly but the number of sessions should be configurable.

**Additional context**
Some times HTTP status could be 522, but anyway it's interuptted.
![64FED39C-7EB1-4441-94F8-F3835DC949C0](https://user-images.githubusercontent.com/4640483/90343479-397ac800-e043-11ea-8e21-e00d488085b3.png)

**Comment by wstyres** [comment]
Seems like an odd bug, 404 not found indicates that the repository is not present and 522 indicates a timeout where the repo did not respond fast enough. There is a configurable timeout option in Zebra's settings, maybe increasing that would result in higher success?

**Comment by CaiMiao** [comment]
Nope.
As I said, excessive network load could crash my VPN tunnel via jetsam, but any connections established before would not fallback, means they all still expected for the tunnel to go.
At last they may split to 2 behaviors (404 or 522) but all the cause is because of the jetsam kill, and the kill is because of the "high" network->RAM load.
As all connections is consider dead when jetsamed, increasing the wait time won't help. They are dead for good.

Currently I can stop the startup repo update, and update each repo manually without problem. But yeah, this would be great effort. The only salvation is reducing network load to avoid jetsam kill.

**Comment by wstyres** [comment]
Alright, I'll take a look about how a sequential refresh could work.
